Src/CSharpier/GroupIdGenerator.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+
+namespace CSharpier
+{
+    public static class GroupIdGenerator
```

Comment:
I'm not thrilled with this class.
I made it static because if we start to split apart printer, it is probably going to move towards static methods.
If we have to instantiate a bunch of different classes in each print method that is going to get obnoxious.
How this gets used also feels a little fragile, but I wasn't sure how else to get a groupId inside of BlockSyntax.
Although, maybe I should just add a method overload like so.... yeah duh. Possibly naming it better while I'm at it.
```c#
private Doc PrintBlockSyntax(BlockSyntax node) => PrintBlockSyntax(node, null);

private Doc PrintBlockSyntax(BlockSyntax node, string? groupId)
{
...
```